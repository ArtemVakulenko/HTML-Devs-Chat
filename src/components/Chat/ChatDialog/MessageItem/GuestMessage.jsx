import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import Button from 'src/components/_common/Button';
import { icons } from 'constants';
import { 
    StGuestMessageContainer, 
    StGuestMessageItem,
    StGuestName,
    StGuestMessageContent,
    StDate,
   } from './styled';

const GuestMessage = ({ authorPic, authorName, content, date, onClick }) => {
    const { colors, theme } = useTheme();
    return (
        <StGuestMessageContainer>
        <Button icon={authorPic || icons.defaultUserIcon} onClick={onClick}/>
        <StGuestMessageItem colors={colors} theme={theme}>
          <StGuestName colors={colors} theme={theme}>
            {authorName}
          </StGuestName>
          <StGuestMessageContent colors={colors} theme={theme}>
            {content}
          </StGuestMessageContent>
          <StDate colors={colors} theme={theme}>
            {date}
          </StDate>
        </StGuestMessageItem>
        </StGuestMessageContainer>
    );
};

GuestMessage.propTypes = {
    authorPic: PropTypes.string,
    authorName: PropTypes.string,
    content: PropTypes.string,
    date: PropTypes.string,
    onClick: PropTypes.func,
};

export default GuestMessage;
