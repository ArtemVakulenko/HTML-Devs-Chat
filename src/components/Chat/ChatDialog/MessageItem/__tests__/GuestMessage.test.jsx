import React from 'react';
import configureStore from 'redux-mock-store';
import GuestMessage from '../GuestMessage';
import { shallowSmart } from '../../../../../__tests__/testHelper';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({ mainPage: {
  sidebar: {
    search: '',
    rooms: [],
  },
  chat: {
    messages: [],
    roomId: '',
    messageValue: '',
    activeRoom: {},
  },
  user: {
    id: '',
    login: '',
    name: '',
    lastName: '',
    img: '',
    hobbie: '',
    city: '',
    age: '',
  },
} });

jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('GuestMessage', () => {
  describe('123', () => {
    const props = {
      message: {},
      onClick: () => jest.fn(),
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<GuestMessage {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
  });
});
