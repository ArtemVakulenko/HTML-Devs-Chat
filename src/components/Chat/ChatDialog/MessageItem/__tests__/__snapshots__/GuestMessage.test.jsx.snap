// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestMessage 123 Should match snapshot 1`] = `"<div class=\\"styled__StGuestMessageContainer-egGbkY fdgoPp\\"><button class=\\"styled__StIcon-chPeWD bvoJqX\\"></button><div class=\\"styled__StGuestMessageItem-gZvHLK libSbh\\"><div class=\\"styled__StGuestName-gPkMPn jEiFwH\\"></div><div class=\\"styled__StGuestMessageContent-jIOva dGqMCY\\"></div><div class=\\"styled__StDate-dCFtOY fvsCbz\\"></div></div></div>"`;

exports[`LoginPage 123 Should match snapshot 1`] = `"<div class=\\"styled__StGuestMessageContainer-egGbkY fdgoPp\\"><button class=\\"styled__StIcon-chPeWD bvoJqX\\"></button><div class=\\"styled__StGuestMessageItem-gZvHLK libSbh\\"><div class=\\"styled__StGuestName-gPkMPn jEiFwH\\"></div><div class=\\"styled__StGuestMessageContent-jIOva dGqMCY\\"></div><div class=\\"styled__StDate-dCFtOY fvsCbz\\"></div></div></div>"`;
